### DISCUSSION TITLE
Detailed Routing abort on libc.so.6

### DISCUSSION THREAD

**Discussion by marcopoles** [The-OpenROAD-Project/OpenROAD#4096] [Q&A]
Hi ,
I'm working on tsmc 40nm porting and the flow stops to work on detailed routing with this error:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/24896919/7c4d6959-f126-4b62-940f-2b3ec59e3d26)

Do you have any suggestion about the problem?

Thanks

PS: working on not open source PDK I can't submit a testcase

**Comment by maliberty** [comment]
@osamahammad21 any guesses?

**Comment by osamahammad21** [comment]
@marcopoles Could you build openroad in DEBUG mode so that I can locate exactly at which line does this happen?

**Reply by marcopoles** [reply]
Could you please let me know the command to run? Thanks

**Reply by osamahammad21** [reply]
If you are building from ORFS, then:
`./build_openroad.sh --local --nice --openroad-args "-DCMAKE_BUILD_TYPE=DEBUG"`
If you are only building OpenROAD, then in the build directory:
```
cmake .. -DCMAKE_BUILD_TYPE=DEBUG
make openroad`
```

**Comment by marcopoles** [comment]
Hi, here the new output:
<img width="883" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD/assets/24896919/bcd2041d-b033-47e4-a166-9304b0c63bfc">

**Reply by marcopoles** [reply]
Running debug in openroad -gui:
I write following commands on tcl row:
```
detailed_route_debug -dr
detailed_route

```
When the yellow button is available, I click it and the program crash. The terminal shows lines below:
<img width="857" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD/assets/24896919/656d2117-cb77-464f-a481-edfb03ea161e">

**Reply by osamahammad21** [reply]
By any chance, is there a via defined between the masterslice layer and the first routing layer in your PDK?

**Reply by marcopoles** [reply]
Hi Osama, my PDK has the definition of connection between masterslice and metal1. Commenting this section (not present for example on nangate45), the routing works!
Thanks for your suggestion!
Now I have the following error probably on RC extraction and I'm investigating on it (for example if I miss some files):

<img width="795" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD/assets/24896919/9552d9a3-4d31-4228-b549-c6390e9b4b2b">

**Reply by marcopoles** [reply]
In the config.mk I commented the line
`export RCX_RULES=...`
to have final gds. 
I see that the RCX_RULES file has to be generated following the "Extraction Rules Generation Flow for OpenRCX", so I have to convert patterns.def into gds and start a commercial tool to extract parasitics into the spef file, right?

**Reply by maliberty** [reply]
You do need a golden result by whatever means (some tools take DEF directly).  It it good to use a separate discussion for unrelated issues.

**Comment by osamahammad21** [comment]
@refaay Could you start a PR to ignore via definitions that include masterslice layers in DRT?

**Comment by refaay** [comment]
Submitted PR #4149 to ignore masterslice vias in drt.
PR #4149 merged.
